code/game/objects/items/weapons/implants/implant.dm

```diff
@@ -1,23 +1,134 @@
+	var/trigger_causes
+	/// Whether this implant has already triggered on death or not, to prevent it firing multiple times.
+	var/has_triggered_on_death = FALSE
 
-/obj/item/implant/proc/trigger(emote, mob/source, force)
+/obj/item/implant/proc/unregister_emotes()
+	if(imp_in && LAZYLEN(trigger_emotes))
+		for(var/emote in trigger_emotes)
+			UnregisterSignal(imp_in, COMSIG_MOB_EMOTED(emote))
+
+/**
+ * Set the emote that will trigger the implant.
+ * * user - User who is trying to associate the implant to themselves.
+ * * emote_key - Key of the emote that should trigger the implant.
+ * * on_implant - Whether this proc is being called during the implantation of the implant.
+ * * silent - If true, the user won't get any to_chat messages if an implantation fails.
+ */
+/obj/item/implant/proc/set_trigger(mob/user, emote_key, on_implant = FALSE, silent = TRUE)
+
+	if(imp_in != user)
+		return FALSE
+
+	if(!emote_key)
+		return FALSE
+
+	if(LAZYIN(trigger_emotes, emote_key) && !on_implant)
+		if(!silent)
+			to_chat(user, "<span class='warning'> You've already registered [emote_key]!")
+		return FALSE
+
+	if(emote_key == "me" || emote_key == "custom")
+		if(!silent)
+			to_chat(user, "<span class='warning'> You can't trigger [src] with a custom emote.")
+		return FALSE
+
+
+	if(!(emote_key in user.usable_emote_keys(trigger_causes & IMPLANT_EMOTE_TRIGGER_INTENTIONAL)))
+		if(!silent)
+			to_chat(user, "<span class='warning'> You can't trigger [src] with that emote! Try *help to see emotes you can use.</span>")
+		return FALSE
+
+	if(!(emote_key in user.usable_emote_keys(trigger_causes & IMPLANT_EMOTE_TRIGGER_UNINTENTIONAL)))
+		CRASH("User was given an implant for an unintentional emote that they can't use.")
+
+	LAZYADD(trigger_emotes, emote_key)
+	RegisterSignal(user, COMSIG_MOB_EMOTED(emote_key), .proc/on_emote)
+
+/obj/item/implant/proc/on_emote(mob/living/user, datum/emote/fired_emote, key, emote_type, message, intentional)
+	SIGNAL_HANDLER
+
+	if(!implanted || !imp_in)
+		return
+
+	if(!(intentional && (trigger_causes & IMPLANT_EMOTE_TRIGGER_INTENTIONAL)) && !(!intentional && (trigger_causes & IMPLANT_EMOTE_TRIGGER_UNINTENTIONAL)))
+		return
+
+	add_attack_logs(user, user, "[intentional ? "intentionally" : "unintentionally"] [src] was [intentional ? "intentionally" : "unintentionally"] triggered with the emote [fired_emote].")
+
+	emote_trigger(key, user, intentional)
+
+/obj/item/implant/proc/on_death(mob/source, gibbed)
+	SIGNAL_HANDLER
+
+	if(!implanted || !imp_in)
+		return
+
+	if(gibbed & trigger_causes & IMPLANT_TRIGGER_NOT_WHEN_GIBBED)
+		return
+
+	// This should help avoid infinite recursion for things like dust that call death()
+	if(has_triggered_on_death && (trigger_causes & IMPLANT_TRIGGER_DEATH_ONCE))
+		return
+
+	has_triggered_on_death = TRUE
+
+	add_attack_logs(source, source, "had their [src] implant triggered on death.")
+
+	death_trigger(source, gibbed)
```

Comment:
Any reason for so much spacing between these?

Reply:
Just habits of mine tending towards more whitespace than less. I'll go through and tidy up a bit.